src/Controller/Component/AuthComponent.php

```diff
@@ -745,25 +764,27 @@ protected function _getUser()
      */
     public function redirectUrl($url = null)
     {
-        if ($url !== null) {
-            $redir = $url;
-            $this->storage()->redirectUrl($redir);
-        } elseif ($redir = $this->storage()->redirectUrl()) {
-            $this->storage()->redirectUrl(false);
+        $redirectUrl = $this->request->query(static::QUERY_STRING_REDIRECT);
+        if ($redirectUrl && (substr($redirectUrl, 0, 1) !== '/')) {
+            $redirectUrl = null;
+        }
```

Comment:
@markstory This patch removes the code from saving referrer to session so trying to read it back doesn't make sense. The referrer isn't persisted for a long duration so i don't think it would cause any issues upgrading.